### PR TITLE
perf: cache parser file list for performance

### PR DESCRIPTION
**Pull request by vigoux**
This speeds up starting time by quite a lot.

### CODE REVIEW

**Comment by steelsojka** [comment]
Do we want a method to invalidate this cache? What after a user runs `TSInstall lua`. Wouldn't this be cached to have no parsers?

**Comment by vigoux** [comment]
Maybe we can invalidate the cache in this case

**Comment by vigoux** [comment]
I just started to think about it, it will be fine if someone installs or updates a parser, as we are only caching file paths and actual parser.

**Comment by vigoux** [comment]
Invalidating the cache is done every time a parser is installed, but not when it is updated, as this does not change the path of the parser.

File: lua/nvim-treesitter/parsers.lua [hunk 1]
```diff
@@ -284,11 +284,25 @@ function M.get_parser_configs()
...
+function M.reset_cache()
+  parser_files = setmetatable({}, {
+    __index = function(tbl, key)
+      rawset(tbl, key, api.nvim_get_runtime_file('parser/' .. key .. '.*', false))
+      return rawget(tbl, key)
+    end
+  })
+end
+
+M.reset_cache()
```
Comment (kyazdani42) [inline lua/nvim-treesitter/parsers.lua:298]: should we call this `init_parser_cache()` ? i'm not too sure, maybe it doesn't really matter.

**Pull request merged**